src/components/TagDropdown.tsx

```diff
@@ -0,0 +1,156 @@
+//smth
+
+//Needs to be added:
+//1. When user is deleting word, filtered list goes back to normal
+//2. Input text field goes over "Add tag" if the tag is really really long
+
+interface TagDropdownProps {
+    tags: string[];
+}
+
+export default function TagDropdown(props: TagDropdownProps) {
+    const [display, setDisplay] = useState(false);
+    const [selectedTags, setSelectedTags] = useState<string[]>([]);
+    const [searchInput, setSearchInput] = useState("");
+    const [filteredTags, setFilteredTags] = useState(props.tags);
+
+    const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const input_text: string = event.target.value;
+        event.preventDefault();
+        setSearchInput(input_text);
+        if (input_text.length > 0) {
+            setFilteredTags(
+                props.tags.filter((tag) => {
+                    return tag.match(input_text);
+                })
+            );
+        } else {
+            setFilteredTags(props.tags);
+        }
+    };
+
+    // Handles actions when 'Enter' key is pressed
+    const handleEnter = (key_event: React.KeyboardEvent) => {
+        if (key_event.key == "Enter") {
+            addTag(searchInput);
+        }
+    }
+
+    const handleAdd = () => {
+        addTag(searchInput);
+    }
+
+    // Controls display of dropdown
+    const handleClick = () => {
+        if (!display) {
+            setDisplay(true);
+        }
+    };
+
+    // Controls tag selection
+    function handleSelectedTags(tag: string) {
+        // if tag is already selected, deselect it and remove from selectedTags
+        if (selectedTags.includes(tag)) {
+            const toRemove = selectedTags.indexOf(tag);
+            setSelectedTags(() => [
+                ...selectedTags.slice(0, toRemove),
+                ...selectedTags.slice(toRemove + 1),
+            ]);
+        } else {
+            setSelectedTags([...selectedTags, tag]);
+        }
+    }
+
+    function deselectTag(tag: string) {
+        const toRemove = selectedTags.indexOf(tag);
+        setSelectedTags(() => [
+            ...selectedTags.slice(0, toRemove),
+            ...selectedTags.slice(toRemove + 1),
+        ]);
+    }
+
+    function addTag(tag: string) {
+        // If tag is in props.tags, add tag to selectedTags array
+        // Else (new tag), add tag to props.tags (large tag array), probably requires request to db, then add tag to selectedTags
+        // ?? QUESTION: do we actually want a New tag to be added every time someone hits enter
+        if (!props.tags.includes(tag)) {
+            props.tags.push(tag);
+            // console.log("pushed tag");
+            // console.log(props.tags);
+        } 
+        setSelectedTags([...selectedTags, tag]);
+        // clear search bar
+        setSearchInput("");
+    }
+
+    return (
+        <div className="flex flex-col p-4 gap-[10px]">
+            <div className="border-2 relative">
```

Comment:
This is super minor but could we remove the border from here? With the border of the input being rounded, having this outer border can look a bit off at the corners:
![2024-01-10_12-25](https://github.com/JumboCode/theatre-at-first/assets/58123848/089c25d7-7c27-4b42-856e-7da78a9a82ce)